radix-engine/src/engine/node.rs

```diff
@@ -8,7 +8,7 @@ pub enum HeapRENode {
     Bucket(Bucket),
     Proof(Proof),
     AuthZone(AuthZone),
-    Vault(Vault),
+    Vault(VaultRuntimeSubstate),
```

Comment:
Can we have a structure for Vault?
```
struct Vault {
    substate: VaultRuntimeSubstate
}
```
I'm imagining we will have multiple states for `Vault` node.

Reply:
Cool